java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -128,17 +128,25 @@ public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers
   @Override
   public List<ArrowBuf> getFieldBuffers() {
     List<ArrowBuf> result = new ArrayList<>(2);
-    validityBuffer.readerIndex(0);
-    validityBuffer.writerIndex(getValidityBufferSizeFromCount(valueCount));
-    offsetBuffer.readerIndex(0);
-    offsetBuffer.writerIndex((valueCount + 1) * OFFSET_WIDTH);
-
+    setReaderAndWriterIndex();
     result.add(validityBuffer);
     result.add(offsetBuffer);
 
     return result;
   }
 
+  private void setReaderAndWriterIndex() {
```

Comment:
Can this be lifted to base class?

Reply:
Probably but I thought BaseRepeatedValueVector on itself is of no use -- at least doesn't represent a materialized field. Moreover, even if we move to base class, we still have to set reader and writer index for validity buffer in sub class since the validity buffer is a property of list vector and not BaseRepeatedValueVector.

Reply:
OK